src/engine/queues/queue.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  QueueInterface,
+  QueueElementInterface,
+  QueueDOMElementInterface
+} from './../interfaces';
+
+export class BaseQueue implements QueueInterface {
+  name: string;
+  protected items: Array<any>;
+  private refCallbacks: Array<{callback:Function;context:Object}> = [];
```

Comment:
Likely we can use a `WeakMap` here to hold this about a particular element. Currently we are not evicting these objects so this list will grow unbounded.